Exercises/SetUpKokkos.cmake

```
@@ -0,0 +1,51 @@
+# Convenience macro to warn the user if a GPU backend is enabled
+macro(KokkosTutorials_WarnGPU)
+    if (Kokkos_ENABLE_CUDA OR Kokkos_ENABLE_HIP OR Kokkos_ENABLE_SYCL OR Kokkos_ENABLE_OPENMPTARGET OR Kokkos_ENABLE_HPX)
+        message(WARNING "cmake"
+                "a Kokkos accelerator backend is enabled, it might cause issue with the current program"
+                "Please recompile with only a host backend enabled (e.g. -DKokkos_ENABLE_OPENMP=ON)")
+    endif ()
+endmacro()
+
+# Early return if Kokkos is already set up
+# We do not use Kokkos_FOUND as it is not globally defined
+if (TARGET Kokkos::kokkos)
+    return()
+endif ()
+
+if (NOT CMAKE_BUILD_TYPE)
+    set(default_build_type "RelWithDebInfo")
+    message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+    set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING
+            "Choose the type of build, options are: Debug, Release, RelWithDebInfo and MinSizeRel."
+            FORCE)
+endif ()
+
+# Where to find Kokkos' source code. This might be set by the user.
+# In order to automatically share the download between exercises when they are compiled individually,
+# the default directory is inside the source tree.
+# This might break if the default in source directory is called from multiple cmake instances at the same time.
+
+set(KokkosTutorials_KOKKOS_SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/dep/kokkos" CACHE PATH "Where Kokkos sources are located")
+
+find_package(Kokkos CONFIG)
```

```suggestion
find_package(Kokkos QUIET)
```

With `CONFIG`, you will get this:

- Kokkos installed:
  ```text
  -- Enabled Kokkos devices: SERIAL;CUDA
  -- kokkos_launch_compiler is enabled globally. C++ compiler commands with -DKOKKOS_DEPENDENCE will be redirected to the appropriate compiler for Kokkos
  ```
- Kokkos not installed:
  ```text
  -- Could NOT find Kokkos (missing: Kokkos_DIR)
  -- Setting default Kokkos CXX standard to 17
  -- Kokkos version: 4.3.1
  <snip>
  ```
  I saw tutorial attendants being confused about the first line.

With `QUIET`, you will get this:

- Kokkos installed:
  ```text
  -- Enabled Kokkos devices: SERIAL;CUDA
  -- kokkos_launch_compiler is enabled globally. C++ compiler commands with -DKOKKOS_DEPENDENCE will be redirected to the appropriate compiler for Kokkos
  ```
- Kokkos not installed:
  ```text
  -- Setting default Kokkos CXX standard to 17
  -- Kokkos version: 4.3.1
  <snip>
  ```

So, `QUIET` seems to be more appropriate.